SalesforceCommerceCloud/pwa-kit
# Fix translation scripts to adapt multi site implementation

# Description
As we implement multi-site feature, we move the `l10n` config from package json to `pwa-kit.config.json`. As a result, some commands for translations are not working.
 
Previously, we have a npm script `formatjs extract 'app/**/*.{js,jsx}' --out-file app/translations/$npm_package_l10n_defaultLocale.json --id-interpolation-pattern [sha512:contenthash:base64:6]`, which will extract default messages and output the outcome into a file name created from npm config `defaultLocale` in `package.json`. This script looks dynamic enough to allow dev to extract messages into their own default locale json file, but it does not take into account all of our default messages are written in English. Image if someone change that locale to `it-IT` and ran the script withouth replace any default messages, they end up having a `it-IT.json` that has English in it.
 
In this sense, the `defaultLocale` has to be _almost_ static. In other words, it has to be `en_xx` and should not be dependent as a config value. 

As an outcome of team discussion, we've decided to extract messages and compile messages to `en-US.json` as American English is what we use for all default messages, and we will have `RefArch` as the default site id because `en-US` is the default locale from BM. 


# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR
- delete `en-GB.json` and `en-US.json`
- run `npm run extract-default-translations` 
- en-GB.json and en-US.json should be created and have all the default messages from the project

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
